celery/canvas.py

```diff
@@ -1808,6 +1810,23 @@ def link(self, callback):
         return callback
 
     def link_error(self, errback):
+        if self.app.conf.task_allow_error_cb_on_chord_header:
+            # self.tasks can be a list of the chord header workflow.
+            if isinstance(self.tasks, list):
+                for task in self.tasks:
+                    task.link_error(errback)
+            else:
+                self.tasks.link_error(errback)
+        else:
+            warnings.warn(
+                "task_allow_error_cb_on_chord_header=False is pending deprecation in "
```

Comment:
so we are introducing task_allow_error_cb_on_chord_header=False to be removed in a future version? what alternative do we have in mind?

Reply:
Currently to make the Enabled state the default when the previous behavior will be fully deprecated. But this is subject to change.